diff --git a/reset_sx1301.sh b/reset_sx1301.sh
new file mode 100755
index 0000000..d399ed7
--- /dev/null
+++ b/reset_sx1301.sh
@@ -0,0 +1,60 @@
+#!/bin/sh
+
+# This script is intended to be used on 1GATE LoRaWAN Basic and Premium Gateway, it performs
+# the following actions:
+#
+#       - export/unexport GPIO25 used to reset the SX1301 chip
+#
+# Usage examples:
+#
+#       ./reset_sx1301.sh stop
+#       ./reset_sx1301.sh start
+
+# The reset pin of SX1301 is wired with RPi GPIO25
+
+    SX1301_RESET_PIN=25
+
+echo "Accessing concentrator reset pin through GPIO$SX1301_RESET_PIN..."
+
+WAIT_GPIO() {
+    sleep 0.1
+}
+
+sx1301_init() {
+    # setup GPIO
+    echo "$SX1301_RESET_PIN" > /sys/class/gpio/export; WAIT_GPIO
+
+    # set GPIO as output
+    echo "out" > /sys/class/gpio/gpio$SX1301_RESET_PIN/direction; WAIT_GPIO
+
+    # write output for SX1301 reset
+    echo "1" > /sys/class/gpio/gpio$SX1301_RESET_PIN/value; WAIT_GPIO
+    echo "0" > /sys/class/gpio/gpio$SX1301_RESET_PIN/value; WAIT_GPIO
+
+    # set GPIO as input
+    echo "in" > /sys/class/gpio/gpio$SX1301_RESET_PIN/direction; WAIT_GPIO
+}
+
+sx1301_term() {
+    # cleanup GPIO
+    if [ -d /sys/class/gpio/gpio$SX1301_RESET_PIN ]
+    then
+        echo "$SX1301_RESET_PIN" > /sys/class/gpio/unexport; WAIT_GPIO
+    fi
+}
+
+case "$1" in
+    start)
+    sx1301_term
+    sx1301_init
+    ;;
+    stop)
+    sx1301_term
+    ;;
+    *)
+    echo "Usage: $0 {start|stop}"
+    exit 1
+    ;;
+esac
+
+exit 0
diff --git a/semtech/.start.sh.swp b/semtech/.start.sh.swp
new file mode 100644
index 0000000..918956b
Binary files /dev/null and b/semtech/.start.sh.swp differ
diff --git a/semtech/install_semtech.sh b/semtech/install_semtech.sh
new file mode 100755
index 0000000..8c8386b
--- /dev/null
+++ b/semtech/install_semtech.sh
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+echo "Install Semtech"
+
+# Stop on the first sign of trouble
+set -e
+
+# Create the required install directory and make sure it is clean
+INSTALL_DIR="/opt/semtech"
+if [ ! -d "$INSTALL_DIR" ]; then mkdir $INSTALL_DIR; fi
+pushd $INSTALL_DIR
+rm -rf *
+
+# Build lora gateway libraries
+if [ ! -d lora_gateway ]; then
+    git clone https://github.com/Lora-net/lora_gateway.git
+    pushd lora_gateway
+else
+    pushd lora_gateway
+    git reset --hard
+    git pull
+fi
+
+make clean all
+
+popd
+
+# Build packet forwarder
+if [ ! -d packet_forwarder ]; then
+    git clone https://github.com/Lora-net/packet_forwarder.git
+    pushd packet_forwarder
+else
+    pushd packet_forwarder
+    git pull
+    git reset --hard
+fi
+
+make clean all
+
+popd
+
+# Symlink
+if [ ! -d bin ]; then mkdir bin; fi
+if [ -f ./bin/lora_pkt_fwd ]; then rm ./bin/lora_pkt_fwd; fi
+ln -s $INSTALL_DIR/packet_forwarder/lora_pkt_fwd/lora_pkt_fwd ./bin/lora_pkt_fwd
+cp -f ./packet_forwarder/lora_pkt_fwd/cfg/global_conf.json.PCB_E286.EU868.basic ./bin/global_conf.json
+cp -f ./packet_forwarder/lora_pkt_fwd/local_conf.json ./bin/local_conf.json
+
+# Reset gateway ID based on MAC
+./packet_forwarder/lora_pkt_fwd/update_gwid.sh ./bin/local_conf.json
+
+echo "Installation completed."
diff --git a/semtech/semtech.service b/semtech/semtech.service
new file mode 100644
index 0000000..328a83d
--- /dev/null
+++ b/semtech/semtech.service
@@ -0,0 +1,12 @@
+[Unit]
+Description=Semtech LoRa Gateway
+
+[Service]
+WorkingDirectory=/opt/semtech/bin/
+ExecStart=/opt/semtech/bin/start.sh
+SyslogIdentifier=semtech
+Restart=on-failure
+RestartSec=5
+
+[Install]
+WantedBy=multi-user.target
diff --git a/semtech/start.sh b/semtech/start.sh
new file mode 100755
index 0000000..a2c0c3b
--- /dev/null
+++ b/semtech/start.sh
@@ -0,0 +1,20 @@
+#! /bin/bash
+
+# Reset SX1301
+# GPIO Header pin 22 / GPIO25
+gpio -1 mode 22 out
+gpio -1 write 22 0
+sleep 0.1
+gpio -1 write 22 1
+sleep 0.1
+gpio -1 write 22 0
+sleep 0.1
+
+# Test the connection, wait if needed.
+while [[ $(ping -c1 google.com 2>&1 | grep " 0% packet loss") == "" ]]; do
+  echo "[LoRa Gateway]: Waiting for internet connection..."
+  sleep 30
+  done
+
+# Fire up the forwarder.
+/opt/semtech/bin/lora_pkt_fwd
diff --git a/update_hostname.sh b/update_hostname.sh
new file mode 100755
index 0000000..47cca73
--- /dev/null
+++ b/update_hostname.sh
@@ -0,0 +1,15 @@
+!/bin/sh
+
+# This script is a helper to update the hostname of the gateway
+
+NEW_HOSTNAME=$(ip link show eth0 | awk '/ether/ {print $2}' | awk -F\: '{print "1gate-"$1$2$3$4$5$6}')
+#NEW_HOSTNAME=${NEW_HOSTNAME^^} #toupper
+
+CURRENT_HOSTNAME=$(hostname)
+
+echo "Updating hostname to '$NEW_HOSTNAME'..."
+hostname $NEW_HOSTNAME
+echo $NEW_HOSTNAME > /etc/hostname
+sed -i "s/$CURRENT_HOSTNAME/$NEW_HOSTNAME/" /etc/hosts
+
+exit 0
